website/docs/r/network_pool.html.markdown

```diff
@@ -0,0 +1,245 @@
+      name = "NsxTDVS"
+    }
+    range_id {
+      start_id = 101
+      end_id   = 200
+    }
+  }
+}
+```
+## Example Usage 3 - Type "PORTGROUP_BACKED"
+
+```hcl
+data "vcd_vcenter" "vc1" {
+  name = "vc1"
+}
+
+resource "vcd_network_pool" "npool" {
+  name                = "my-pg-network-pool"
+  description         = "New Port Group network pool"
+  network_provider_id = data.vcd_vcenter.vc1.id
+  type                = "PORTGROUP_BACKED"
+
+  backing {
+    port_group {
+      name = "TestbedPG"
+    }
+  }
+}
+```
+
+## Example Usage 4 Retrieving backing elements
+
+The elements needed as backing for a network pool can be retrieved using [`vcd_resource_list`](/providers/vmware/vcd/latest/docs/data_sources/resource_list), as in the example below
+
+```hcl
+data "vcd_nsxt_manager" "mgr" {
+  name = "nsxManager1"
+}
+
+data "vcd_vcenter" "vc1" {
+  name = "vc1"
+}
+
+data "vcd_resource_list" "tz" {
+  name          = "tz"
+  resource_type = "vcd_nsxt_transport_zone"
+  parent        = data.vcd_nsxt_manager.mgr.name
+}
+
+data "vcd_resource_list" "pg" {
+  name          = "pg"
+  resource_type = "vcd_importable_port_group"
+  parent        = data.vcd_vcenter.vc1.name
+}
+
+data "vcd_resource_list" "ds" {
+  name          = "ds"
+  resource_type = "vcd_distributed_switch"
+  parent        = data.vcd_vcenter.vc1.name
+}
+
+output "tzs" {
+  value = data.vcd_resource_list.tz.list
+}
+
+output "pgs" {
+  value = data.vcd_resource_list.pg.list
+}
+
+output "ds" {
+  value = data.vcd_resource_list.ds.list
+}
+```
+
+-> Note: the lists provided as `vcd_resource_list` output are volatile: they only exist for items that have not been used
+in a network pool. Once they have been assigned, they cease to be shown. As such, it is not a good idea to use
+`vcd_resource_list` as direct source for one or more network pools: at the first `plan`, terraform would propose
+to remove the network pool, as the element is not shown in the list anymore.
+
+## Example Usage 5 - nameless backing retrieval
+
+If we don't have preference about which of the elements we will use as backing for the network pool, we could let
+the system pick the first available. This could be a good idea when we know that there is only one element available, or 
+we know that all elements have similar capabilities.
+If we are in these circumstances, we could avoid some details and skip the definition of the backing elements, but we
+must specify the value of `backing_components_use_constraint` as either `use-when-only-one` or `use-first-available`.
+
+```hcl
+data "vcd_nsxt_manager" "mgr" {
+  name = "nsxManager1"
+}
+
+data "vcd_vcenter" "vc1" {
+  name = "vc1"
+}
+
+resource "vcd_network_pool" "npool-1" {
+  name                = "new-network-pool"
+  description         = "network pool without explicit transport zone"
+  network_provider_id = data.vcd_vcenter.vc1.id
+  type                = "PORTGROUP_BACKED"
+
+  backing_components_use_constraint = "use-when-only-one"
+
+  backing {
+  }
```

Comment:
This section looks really extra when automatic backing selection is used. Can we make it work without it:

```
resource "vcd_network_pool" "npool-1" {
  name                = "new-network-pool"
  description         = "network pool without explicit transport zone"
  network_provider_id = data.vcd_vcenter.vc1.id
  type                = "PORTGROUP_BACKED"

  backing_components_use_constraint = "use-when-only-one"
}
```

Reply:
Not really. It's an example of how to use the automatic backing selection.

Reply:
Updated docs and tests